sdk/python/approzium/psycopg2.py

```diff
@@ -1,18 +1,51 @@
 import psycopg2
 import select
 import socket
+from socket import ntohl, htonl
 import struct
 import logging
-from ctypes import cdll, create_string_buffer
+from sys import getsizeof
+from ctypes import cdll, create_string_buffer, string_at, memmove
 from ctypes.util import find_library
 from .socketfromfd import fromfd
 from .authenticator import get_hash
 from .misc import read_int32_from_bytes
 
+
 libpq = cdll.LoadLibrary("libpq.so.5")
 libssl = cdll.LoadLibrary(find_library("ssl"))
 
 
+def set_connection_sync(pgconn):
+    mem = bytearray(string_at(id(pgconn), getsizeof(pgconn)))
+    sizeofint = struct.calcsize("@i")
+    sizeoflong = struct.calcsize("@l")
+
+    def addressofint(number, mem=mem):
+        int_bytes = struct.pack("@i", number)
+        return mem.find(int_bytes)
+
+    def intataddress(address):
+        return struct.unpack("@i", mem[address : address + sizeofint])[0]
+
+    # as a check, we check server and protocol version numbers, which succeed
+    # the async value in the psycopg connection struct
+    server_version_addr = addressofint(pgconn.server_version)
+    # check that there is only one match for that value
+    assert addressofint(pgconn.server_version, mem[server_version_addr+sizeofint:]) == -1
```

Comment:
if I'm not mistaken, these statements will raise generic AssertionErrors -- is there any reason we prefer this over more descriptive messaging/graceful handling?